packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts

```diff
@@ -0,0 +1,440 @@
+    { credentialFormats }: FormatCreateRequestOptions<JsonLdCredentialFormat>
+  ): Promise<FormatCreateReturn> {
+    const jsonLdFormat = credentialFormats?.jsonld
+
+    const format = new CredentialFormatSpec({
+      format: JSONLD_VC_DETAIL,
+    })
+
+    if (!jsonLdFormat) {
+      throw new AriesFrameworkError('Missing jsonld payload in createRequest')
+    }
+
+    // FIXME: validating an interface doesn't work.
+    const jsonLdCredential = new JsonLdCredential(jsonLdFormat)
+    MessageValidator.validateSync(jsonLdCredential)
+
+    // FIXME: SignCredentialOptions doesn't follow RFC0593
+    const attachment = this.getFormatData(jsonLdFormat, format.attachId)
+
+    return { format, attachment }
+  }
+
+  public async processRequest(agentContext: AgentContext, { attachment }: FormatProcessOptions): Promise<void> {
+    // FIXME: SignCredentialOptions doesn't follow RFC0593
+    const requestJson = attachment.getDataAsJson<SignCredentialOptionsRFC0593>()
+
+    if (!requestJson) {
+      throw new AriesFrameworkError('Missing jsonld credential request data payload')
+    }
+
+    // FIXME: validating an interface doesn't work.
+    const jsonLdCredential = new JsonLdCredential(requestJson)
+    MessageValidator.validateSync(jsonLdCredential)
+  }
+
+  public async acceptRequest(
+    agentContext: AgentContext,
+    { credentialFormats, attachId, requestAttachment }: FormatAcceptRequestOptions<JsonLdCredentialFormat>
+  ): Promise<FormatCreateReturn> {
+    const jsonLdFormat = credentialFormats?.jsonld
+
+    // sign credential here. credential to be signed is received as the request attachment
+    // (attachment in the request message from holder to issuer)
+    const credentialRequest = requestAttachment.getDataAsJson<SignCredentialOptionsRFC0593>()
+
+    let verificationMethod = credentialFormats?.jsonld?.verificationMethod
+
+    // FIXME: Need to transform from json to class instance
+    // FIXME: SignCredentialOptions doesn't follow RFC0593
+    // FIXME: Add validation of the jsonLdFormat data (if present)
+    const credentialData = jsonLdFormat ?? credentialRequest
+
+    // FIXME: we're not using all properties from the interface. If we're not using them,
+    // they shouldn't be in the interface.
+    if (!verificationMethod) {
+      verificationMethod = await this.deriveVerificationMethod(
+        agentContext,
+        credentialData.credential,
+        credentialRequest
+      )
+      if (!verificationMethod) {
+        throw new AriesFrameworkError('Missing verification method in credential data')
+      }
+    }
+    const format = new CredentialFormatSpec({
+      attachId,
+      format: JSONLD_VC,
+    })
+
+    const verifiableCredential = await this.w3cCredentialService.signCredential(agentContext, {
+      credential: JsonTransformer.fromJSON(credentialData.credential, W3cCredential),
+      proofType: credentialData.options.proofType,
+      verificationMethod: verificationMethod,
+    })
```

Comment:
We're ignoring most of the options from the RFC (proofPurpose, created, challenge, domain, credentialStatus). If we arent' using these, they shouldn't be in the interface

Reply:
This would entail a change in the W3cCredentialService (signCredential method), would it not?

Do you want me to go ahead and make that change?

Reply:
(Note that `created `and `proofPurpose `are referred to inside this method so presumably we don't want to take those out).

I propose to remove `challenge`, `domain `and `credentialStatus `which are not used.

Reply:
Yes we should probably add them later on, but for now it's fine to leave them out I think